scripts/ci-entrypoint.sh

```diff
@@ -133,26 +133,24 @@ create_cluster() {
 
 install_addons() {
     # Get cluster CIDRs from Cluster object
-    CIDR0=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o=jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[0]}')
-    CIDR1=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o=jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[1]}' 2> /dev/null) || true
-
-    # export the target cluster KUBECONFIG if not already set
-    export KUBECONFIG="${KUBECONFIG:-${PWD}/kubeconfig}"
+    CIDR0=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[0]}')
+    CIDR1=$(${KUBECTL} get cluster "${CLUSTER_NAME}" -o jsonpath='{.spec.clusterNetwork.pods.cidrBlocks[1]}' 2> /dev/null) || true
 
     # wait for the apiserver pod to be Ready.
-    APISERVER_POD=$("${KUBECTL}" get pods -n kube-system -o name | grep apiserver)
-    "${KUBECTL}" wait --for=condition=Ready -n kube-system "${APISERVER_POD}" --timeout=5m
+    timeout --foreground 600 bash -c "while ! ${KUBECTL} --kubeconfig=./${CLUSTER_NAME}.kubeconfig get pods -n kube-system -o name | grep apiserver; do sleep 5; done"
```

Comment:
In my testing I encountered further flakes that originate from grabbing the apiserver pod (L141 below) blindly, without first confirming that it exists. I think I'd like to address that in a separate PR.

My thoughts are: rather than doing this check for a particular grep'd apiserver pod (which I would expect to have unreliable results in a HA control plane scenario where the grep returns more than one line entry), why don't we just  put the below kubectl commands in appropriate retry blocks? That would seem more resilient than trying to identify every possible "apiserver pod isn't stable" symptom, and additionally a validation at one point in time that "apiserver pod is stable" isn't durable, and moments later a kubectl command against it may transiently fail (such as during a > 1 control plane node etcd leader election scenario during cluster bootstrapping).

Reply:
I've removed this change from this PR and scoped these changes just to the persisted workload cluster kubeconfig.